pypit/par/pypitpar.py

```diff
@@ -0,0 +1,2486 @@
+        in the current working directory.
+        
+        .. todo::
+            - Remove default_spectrograph.cfg
+        """
+        # Name of the file to find
+        file_name = '{0}_spectrograph.cfg'.format(key)
+
+        # First try to find the file in the local directory
+        cfg_file = os.path.join(os.getcwd(), file_name)
+        if os.path.isfile(cfg_file):
+            if verbose:
+                print('Found: {0}'.format(cfg_file))
+            return cfg_file
+
+        # Then try to find the file in the distribution
+        pypit_root = _pypit_root_directory()
+        spec_dir = os.path.join(pypit_root, 'pypit', 'config', 'spectrographs')
+        cfg_file = os.path.join(spec_dir, file_name)
+        if os.path.isfile(cfg_file):
+            if verbose:
+                print('Found: {0}'.format(cfg_file))
+            return cfg_file
+
+        # Could not find the file!
+        raise ValueError('Could not find associated configuration file in either the local'
+                         'directory or the pypit source distribution for '
+                         'spectrograph {0}!'.format(key))
+
+    @staticmethod
+    def valid_pipelines():
+        """Return the list of allowed pipelines within pypit."""
+        return [ 'ARMS', 'ARMED' ]
+
+    @staticmethod
+    def allowed_master_options():
+        """Return the allowed handling methods for the master frames."""
+        return [ 'reuse', 'force' ]
+
+    def validate(self):
+        pass
+
+    
+class WavelengthSolutionPar(ParSet):
+    def __init__(self, method=None, lamps=None, detection=None, numsearch=None, nfitpix=None,
+                 IDpixels=None, IDwaves=None):
+
+        # Grab the parameter names and values from the function
+        # arguments
+        args, _, _, values = inspect.getargvalues(inspect.currentframe())
+        pars = dict([(k,values[k]) for k in args[1:]])
+
+        # Initialize the other used specifications for this parameter
+        # set
+        defaults = dict.fromkeys(pars.keys())
+        options = dict.fromkeys(pars.keys())
+        dtypes = dict.fromkeys(pars.keys())
+        descr = dict.fromkeys(pars.keys())
+
+        # Fill out parameter specifications.  Only the values that are
+        # *not* None (i.e., the ones that are defined) need to be set
+        defaults['method'] = 'arclines'
+        options['method'] = WavelengthSolutionPar.valid_methods()
+        dtypes['method'] = str
+        descr['method'] = 'Method to use to fit the individual arc lines.  ' \
+                          '\'fit\' is likely more accurate, but \'simple\' uses a polynomial ' \
+                          'fit (to the log of a gaussian) and is fast and reliable.  ' \
+                          '\'arclines\' uses the arclines python package.' \
+                          'Options are: {0}'.format(', '.join(options['method']))
+
+        options['lamps'] = WavelengthSolutionPar.valid_lamps()
+        dtypes['lamps'] = [str, list]
```

Comment:
Probably should insist on a list.
Even if only a single lamp.

Reply:
Okay, done.  I added a couple of lines of code to deal with a config file that only lists one lamp